cli/provider_cache.go

```diff
@@ -212,36 +174,100 @@ func providerCacheEnvironment(opts *options.TerragruntOptions, cliConfigFile str
 //		}
 //	}
 //
-// This func doesn't change the default CLI config file, only creates a new one at the given path `cliConfigFile`. Ultimately, we can assign this path to `TF_CLI_CONFIG_FILE`.
-func createLocalCLIConfig(opts *options.TerragruntOptions, cliConfigFile string, registryProviderURL *url.URL) error {
+// This func doesn't change the default CLI config file, only creates a new one at the given path `filename`. Ultimately, we can assign this path to `TF_CLI_CONFIG_FILE`.
+//
+// It creates two types of configuration depending on the `cacheRequestID` variable set.
+// 1. If `cacheRequestID` is set, `terraform init` does _not_ use the provider cache directory, the cache server creates a cache for requested providers and returns HTTP status 423. Since for each module we create the CLI config, using `cacheRequestID` we have opprotuenty later retrieve from the cache server exactly those cached providers that were requested by `terraform init` using this configuration.
```

Comment:
```suggestion
// 1. If `cacheRequestID` is set, `terraform init` does _not_ use the provider cache directory, the cache server creates a cache for requested providers and returns HTTP status 423. Since for each module we create the CLI config, using `cacheRequestID` we have the opportunity later retrieve from the cache server exactly those cached providers that were requested by `terraform init` using this configuration.
```

Reply:
Thanks.